### PR TITLE
AKU-203:Document locate action

### PR DESCRIPTION
**Pull request by draperd**
This PR addresses https://issues.alfresco.com/jira/browse/AKU-203. This isn't quite complete pending work on AKU-162 (for link creation) so this service will be updated when link creation is supported.

### CODE REVIEW

File: aikau/src/main/resources/alfresco/renderers/_ItemLinkMixin.js [hunk 1]
```diff
@@ -82,10 +86,10 @@ define(["dojo/_base/declare",
          }
          else
          {
-            if (this.currentItem && this.currentItem.node)
+            if (item && item.node)
             {
-               var jsNode = this.currentItem.jsNode;
-               if (jsNode.isLink && this.currentItem.location.site)
+               var jsNode = item.jsNode;
+               if (jsNode.isLink && item.location.site)
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/renderers/_ItemLinkMixin.js]: Could item.jsNode be `null` or `undefined` here (i.e. causing `jsNode.isLink` to error)?

File: aikau/src/main/resources/alfresco/services/actions/NodeLocationService.js [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+       * @default false
+       */
+      useAikauPages: false,
+
+      /**
+       * @instance
+       * @param {array} args Constructor arguments
+       */
+      constructor: function alfresco_services_actions_NodeLocationService__constructor(args) {
+         lang.mixin(this, args);
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/services/actions/NodeLocationService.js]: Should we be using [`dojo/_base/declare#safeMixin()`](https://dojotoolkit.org/reference-guide/1.10/dojo/_base/declare.html#dojo-base-declare-safemixin) here instead?

File: aikau/src/test/resources/alfresco/services/actions/NodeLocationTest.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            type: "PUBLISH",
+            topic: "ALF_NAVIGATE_TO_PAGE",
+            pos: "last"
+         })
+         .then(function(payload) {
+            assert.isNotNull(payload, "Navigation publication not found");
+            if (payload) {
+               assert.equal(payload.url, "/aikau/page/site/site1/documentlibrary#path=%2Fsome%2Frandom%2Fpath", "Unexpected location");
+            }
+         });
```
Comment (MartinDoyleUK) [inline aikau/src/test/resources/alfresco/services/actions/NodeLocationTest.js]: /like

File: aikau/src/main/resources/alfresco/services/actions/NodeLocationService.js (hunk 2 above)
Comment (draperd) [inline aikau/src/main/resources/alfresco/services/actions/NodeLocationService.js]: Not sure, I've used lang.mixin in all the other service constructors as well. We'd probably want to change all of them.

**Pull request merged**